### PR TITLE
Enable import following in MYPYNOFOLLOW (now MYPYINDUCTOR)

### PR DESCRIPTION
**Pull request by int3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113830
* #113721
* #113722

Skipping importing some packages for now to make this change more
tractable.

For some reason, lintrunner on CI raises errors in all imported `.pyi` files,
even though it doesn't on my local machine. The errors are all from missing
generic types, as the MYPYINDUCTOR config has `disallow_any_generics`
set. I have thus added `disable-error-code` comments to the relevant files,
though I fixed a few that were easy enough.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -1518,7 +1519,7 @@ def get_fake_value(node, tx, allow_non_graph_fake=False):
                 f"constrain_as_value OR constrain_as_size APIs.  {cause}",
                 case_name="constrain_as_size_example",
             )
-        elif isinstance(cause, torch.utils._sympy.value_ranges.ValueRangeError):
```
Comment (int3) [inline torch/_dynamo/utils.py:1521]: somehow having `follow_imports=skip` on `mypy-torch.utils._sympy.*` causes mypy to think `ValueRangeError` doesn't exist in the module it's not inspecting. But the error goes away if we explicitly import it

File: .lintrunner.toml [hunk 2]
```diff
@@ -178,11 +178,10 @@ init_command = [
 ]
 
 [[linter]]
-code = 'MYPYNOFOLLOW'
+code = 'MYPYINDUCTOR'
 include_patterns = [
     'torch/_dynamo/**/*.py',
     'torch/_inductor/**/*.py',
-    'torch/_C/_dynamo/**/*.py',
```
Comment (int3) [inline .lintrunner.toml:185]: this doesn't really do anything, it doesn't match the `.pyi` files in that folder, and there are no `.py` files there

File: torch/_C/_profiler.pyi [hunk 3]
```diff
@@ -58,7 +58,6 @@ class _ExperimentalConfig:
         profiler_measure_per_kernel: bool = ...,
         verbose: bool = ...,
         performance_events: List[str] = ...,
-        privateuse1_config: Dict = ...,
```
Comment (int3) [inline torch/_C/_profiler.pyi:61]: I don't think this is a valid param to the constructor:

https://github.com/pytorch/pytorch/blob/0d6d97d956690b16ba5224f8accf748b90d16108/torch/csrc/profiler/python/init.cpp#L265-L273

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**